src/sif/truckcost.cc

```diff
@@ -126,6 +127,14 @@ class TruckCost : public DynamicCost {
 
   virtual ~TruckCost();
 
+  /**
+   * If we want to penalize HGV access, pretend we're driving a car so hgv=no edges are allowed.
+   * @return  Returns car access if we want to penalize hgv=no or truck access by default.
+   */
+  virtual uint32_t access_mode() const override {
+    return truck_access_mask_;
```

Comment:
`truck_access_mask_` will be `kTruckAccess` by default, but `kAutoAccess` if `hgv_no_access_penalty`  is set in the request

Reply:
let's make this protected